src/main/java/sparksoniq/jsoniq/runtime/iterator/operational/UnaryOperationIterator.java

```diff
@@ -48,9 +48,13 @@ public Item next() {
         if (this._hasNext) {
             this._hasNext = false;
             _child.open(_currentDynamicContext);
-            Item child = _child.next();
+            Item child = null;
```

Comment:
Perfect. I think it is now indeed cleaner, as the first hasNext() value depends on the semantics of the unary iterator.

Could you please add two tests

-() (returns ())
(1, -(), 1) (returns (1,1))

to check that the semantics of the unary minus operator (which does, unlike "instance of", return the empty sequence if the child returns the empty sequence) has no regression in spite of this change in the code?

Also, () instance of string* and () instance of string+ would be good corner case tests.

Thanks a lot! Then it's good to go. I'll mark as approved and you can merge after adding the above tests and checking they work.

Reply:
I now added the tests for UnaryOperation for empty sequence as well.